src/app/3d/qgs3dnavigationwidget.cpp

```diff
@@ -202,5 +202,5 @@ Qgs3DNavigationWidget::Qgs3DNavigationWidget( Qgs3DMapCanvas *parent ) : QWidget
 void Qgs3DNavigationWidget::updateFromCamera()
 {
   // Make sure the angle is between 0 - 359
-  mCompas->setValue( ( int( mParent3DMapCanvas->cameraController()->yaw() ) % 360 + 360 ) % 360 );
+  mCompas->setValue( fmod( fmod( mParent3DMapCanvas->cameraController()->yaw(), 360 ) + 360, 360 ) );
```

Comment:
This looks a little strange to me (the before one!) -- can you see why there's two mods here in the first place? Wouldn't 
 
    mCompas->setValue( fmod( mParent3DMapCanvas->cameraController()->yaw() + 360, 360 );

give the same result?

Reply:
for negative value I think.

Reply:
But that shouldn't make a difference, because eg mod(-270 + 360, 360 ) = 90.

Reply:
indeed...